server/src/main/java/org/opensearch/index/search/stats/SearchStats.java

```diff
@@ -391,6 +450,60 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.humanReadableField(Fields.SUGGEST_TIME_IN_MILLIS, Fields.SUGGEST_TIME, getSuggestTime());
             builder.field(Fields.SUGGEST_CURRENT, suggestCurrent);
 
+            if (requestStatsLongHolder != null) {
+                builder.startObject(Fields.REQUEST);
+
+                builder.humanReadableField(
```

Comment:
Similar to the `StatsHolder` class in `SearchRequestStats`, you can define a holder for the snapshot counters like:

```
class PhaseStatsLongHolder implements Writeable {
  long current;
  long total;
  long timeInMillis;
}
```

This whole block can be:

```
  for (SearchPhaseName phaseName: SearchPhaseName.values()) {
    PhaseStatsLongHolder phaseStats = requestStatsHolder.get(phaseName);
    // Create an object for each phase:
    builder.startObject(phaseName.getName());
    builder.field(Fields.TIME_IN_MILLIS, Fields.QUERY_TIME, new TimeValue(phaseStats.timeInMillis));
    builder.field(Fields.CURRENT, phaseStats.current);
    builder.field(Fields.TOTAL, phaseStats.total);
    builder.endObject();
  }
```

Reply:
Changed. Thanks for suggestion @msfroh !